pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -101,17 +103,50 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if readOnly {
 		readWrite = "READ_ONLY"
 	}
-	attachedDisk := gce.convertDiskToAttachedDisk(disk, readWrite)
 
 	mc := newDiskMetricContext("attach", instance.Zone)
-	attachOp, err := gce.service.Instances.AttachDisk(
-		gce.projectID, disk.Zone, instance.Name, attachedDisk).Do()
+	source := gce.service.BasePath + strings.Join([]string{
+		gce.projectID, "zones", disk.Zone, "disks", disk.Name}, "/")
+	attachOp, err := gce.getAttachDiskOp(
+		disk.Name, disk.Kind, disk.Zone, readWrite, source, diskTypePersistent, instance.Name)
 
 	if err != nil {
 		return mc.Observe(err)
 	}
 
-	return gce.waitForZoneOp(attachOp, disk.Zone, mc)
+	return gce.manager.WaitForZoneOp(attachOp, disk.Zone, mc)
+}
+
+// Returns a version agnostic AttachDisk operation.
+func (gce *GCECloud) getAttachDiskOp(
```

Comment:
Maybe call this "startAttachDiskOp" to make it more clear that this call also starts the operation.